crates/hyperqueue/src/dashboard/ui/screens/home/chart_section.rs

```diff
@@ -0,0 +1,132 @@
+                    Ok(time) => Some((time.as_secs() as f64, state.worker_count as f64)),
+                    Err(_) => None,
+                })
+                .collect();
+
+            let datasets = vec![Dataset::default()
+                .name("workers_connected")
+                .marker(symbols::Marker::Braille)
+                .style(Style::default().fg(Color::White))
+                .data(&worker_counts)];
+
+            let chart = Chart::new(datasets)
+                .block(
+                    Block::default()
+                        .title(Span::styled(
+                            "Worker Connection History",
+                            Style::default()
+                                .fg(Color::White)
+                                .add_modifier(Modifier::BOLD),
+                        ))
+                        .borders(Borders::ALL),
+                )
+                .x_axis(
+                    Axis::default()
+                        .title("time ->")
+                        .style(Style::default().fg(Color::Gray))
+                        .bounds([
+                            get_time_as_secs() as f64
+                                - self.view_size.as_secs() as f64,
+                            get_time_as_secs() as f64,
+                        ]),
+                )
+                .y_axis(
+                    Axis::default()
+                        .style(Style::default().fg(Color::Gray))
+                        .bounds([0.0, self.get_max_workers_in_view().unwrap_or(0) as f64])
+                        .labels(vec![
+                            Span::from(0.to_string()),
+                            Span::from(
+                                (self.get_max_workers_in_view().unwrap_or(0) / 2).to_string(),
+                            ),
+                            Span::from((self.get_max_workers_in_view().unwrap_or(0)).to_string()),
+                        ]),
+                );
+            frame.render_widget(chart, rect);
+        } else {
+            let header_text = vec![Spans::from("No workers connected")];
+            let text_block = Block::default()
+                .borders(Borders::ALL)
+                .style(Style::default().bg(Color::Black));
+            let paragraph = Paragraph::new(header_text)
+                .block(text_block)
+                .alignment(Alignment::Left)
+                .wrap(Wrap { trim: true });
+            frame.render_widget(paragraph, rect);
+        }
+    }
+
+    /// Returns the maximum number of workers connected in the past n seconds specified by view_size
+    fn get_max_workers_in_view(&self) -> Option<u32> {
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map(|current_time| {
+                self.worker_count_vec
+                    .iter()
+                    .filter(|count| {
+                        count.at_time.duration_since(UNIX_EPOCH).unwrap().as_secs()
```

Comment:
You can compare `SystemTime`s directly, you don't have to convert them to seconds.